packages/flutter_availability/lib/src/ui/screens/availability_modification.dart

```diff
@@ -127,111 +116,156 @@ class _AvailabilitiesModificationScreenState
+}
 
-    // TODO(Joey): this structure is defined multiple times, we should create
-    // a widget to handle this consistently
-    var body = CustomScrollView(
-      slivers: [
-        SliverPadding(
-          padding:
-              EdgeInsets.symmetric(horizontal: options.spacing.sidePadding),
-          sliver: SliverList.list(
-            children: [
-              const SizedBox(height: 40),
-              clearSection,
-              if (!_clearAvailability) ...[
-                const SizedBox(height: 24),
-                templateSelection,
-                const SizedBox(height: 24),
-                timeSelection,
-                // TODO(Joey): Not divisible by 4
-                const SizedBox(height: 26),
-                pauseSelection,
-              ],
-            ],
-          ),
-        ),
-        SliverFillRemaining(
-          fillOverscroll: false,
-          hasScrollBody: false,
-          child: Padding(
-            padding: EdgeInsets.symmetric(
-              horizontal: spacing.sidePadding,
-            ).copyWith(
-              bottom: spacing.bottomButtonPadding,
-            ),
-            child: Align(
-              alignment: Alignment.bottomCenter,
-              child: saveButton,
-            ),
-          ),
-        ),
-      ],
-    );
+class _AvailabilitiesModificationScreenLayout extends StatelessWidget {
+  const _AvailabilitiesModificationScreenLayout({
+    required this.dateRange,
+    required this.clearAvailability,
+    required this.onClearSection,
+    required this.selectedTemplates,
+    required this.onTemplateSelected,
+    required this.onTemplatesRemoved,
+    required this.startTime,
+    required this.endTime,
+    required this.onStartChanged,
+    required this.onEndChanged,
+    required this.breaks,
+    required this.onBreaksChanged,
+    required this.sidePadding,
+    required this.bottomButtonPadding,
+    required this.saveButton,
+    required this.onExit,
+  });
+
+  final DateTimeRange dateRange;
+  final bool clearAvailability;
+  // ignore: avoid_positional_boolean_parameters
+  final void Function(bool isChecked) onClearSection;
+
+  final List<AvailabilityTemplateModel> selectedTemplates;
+  final void Function() onTemplateSelected;
+  final void Function() onTemplatesRemoved;
+
+  final TimeOfDay? startTime;
+  final TimeOfDay? endTime;
+  final void Function(TimeOfDay start) onStartChanged;
+  final void Function(TimeOfDay start) onEndChanged;
+
+  final List<BreakViewModel> breaks;
+  final void Function(List<BreakViewModel> breaks) onBreaksChanged;
+
+  final double sidePadding;
+  final double bottomButtonPadding;
+
+  final Widget saveButton;
+
+  final VoidCallback onExit;
+
+  @override
+  Widget build(BuildContext context) {
+    var availabilityScope = AvailabilityScope.of(context);
+    var options = availabilityScope.options;
 
     return options.baseScreenBuilder(
       context,
-      widget.onExit,
-      body,
+      onExit,
+      BasePage(
```

Comment:
extract this to a variable

Reply:
What would be the point of that? Most of this has already been extracted, I see no use at all in moving this to a variable.